rustls/src/crypto/aws_lc_rs/tls12.rs

```diff
@@ -449,17 +437,47 @@ impl Prf for Tls12Prf {
         label: &[u8],
         seed: &[u8],
     ) -> Result<(), Error> {
-        self.for_secret(
-            output,
+        Tls12PrfSecretBorrow(
+            self.0,
             kx.complete_for_tls_version(peer_pub_key, &TLS12)?
                 .secret_bytes(),
-            label,
-            seed,
-        );
+        )
+        .prf(output, label, seed);
         Ok(())
     }
 
+    fn new_secret(&self, secret: &[u8; 48]) -> Box<dyn PrfSecret> {
```

Comment:
Suggest taking an owned array if we're going to store it -- especially for secrets? (Up the chain as relevant.)